extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -675,6 +823,82 @@ private void internalReloadDomains() {
     }
   }
 
+  /**
+   * POC - DEBUG
+   * Only migrate if certain properties don't exist and other conditions.
+   * Else return file metadata.
+   * @param serializableId
+   * @return
+   */
+  public Map<String, Serializable> getFileMetadataHelper( Serializable serializableId ) {
+    Map<String, Serializable> fileMetadata = repository.getFileMetadata( serializableId );
+
+    // check for domain, has not been previously migrated, flag set to migrate
+    if ( isDomain( fileMetadata ) && !hasDatasourceType( fileMetadata ) && canMigrate() ) {
+      fileMetadata = migrateDomain( serializableId, fileMetadata );
+    }
+
+    return  fileMetadata;
+  }
+
+  /**
+   * Determines if file metadata has domain related properties.
+   * @param fileMetadata
+   * @return
+   */
+  boolean isDomain( Map<String, Serializable> fileMetadata ) {
+    return StringUtils.isNotEmpty( (String) fileMetadata.get( PROPERTY_NAME_DOMAIN_ID ) )
+            && StringUtils.equals( TYPE_DOMAIN, (String) fileMetadata.get( PROPERTY_NAME_TYPE ) );
+  }
+
+  /**
+   * Determines if Domain file metadata has property #PROPERTY_NAME_DATASOURCE_TYPE
+   * @param fileMetadata
+   * @return
+   */
+  boolean hasDatasourceType( Map<String, Serializable> fileMetadata ) {
+    return fileMetadata.containsKey( PROPERTY_NAME_DATASOURCE_TYPE );
+  }
+
+  /**
+   * Determines if system settings will allow migration of domain objects.
+   * @return
+   */
+  protected boolean canMigrate() {
+    return BooleanUtils.toBoolean( systemConfig.getProperty( "system.migrate-domains",
+            BooleanUtils.toStringTrueFalse( DEFAULT_MIGRATE_DOMAINS ) ) );
+  }
+
+  /**
+   * Determines if system settings will allow migration of domain objects.
+   * @return
+   */
+  protected int getNumberOfThreads() {
```

Comment:
TODO need to remove, this is dead code PentahoMetadataDomainRepository.java#getNumberOfThreads() .It is already in SessionCachingMetadataDomainRepository

Reply:
see commit: https://github.com/pentaho/pentaho-platform/pull/5124/commits/2277f57c5f89124e2e5b11b6591f46694b98350d